dali/pipeline/executor/executor.h

```diff
@@ -140,6 +140,24 @@ class DLL_PUBLIC Executor : public ExecutorBase, public WorkspacePolicy, public
 
   void SetupOutputQueuesForGraph();
 
+  template <typename Workspace>
+  void DoRun(OperatorBase &op, Workspace &ws) {
+    std::vector<kernels::TensorListShape<>> output_shapes;
```

Comment:
Please make these vectors persistent across as many invocations as possible - e.g. by storing them in a helper object and just calling `clear` before each reuse.

Reply:
Done